### PR TITLE
Implement database design

### PR DESCRIPTION
**Pull request by zanq-mw**
Implemented database design somewhat following this schema
LaunchPad DB V2

Since were using a nosql db i changed a few things. InternshipPosting and NewGradPosting data is now included inside of PostingData. UserPosting is now included in User as savedJobs. I am unsure if we let Notification stand on its own or merge it into User data.

Also example query implemented and results of it can be found by going to localhost:5000/application

Screenshot of db:
image

### CODE REVIEW

File: launchpad_backend/launchpad_server/routes/startup_data.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    "records": [{
+        "notificationId": 1,
+        "subject": "you got hired", 
+        "body": "im lying", 
+        "dateTime": datetime(2020, 5, 17, 7, 30, 54), 
+        "read": True, 
+        "saved": False, 
+        "applicationId": 1
+    }]
+}
```
Comment (zanq-mw) [inline launchpad_backend/launchpad_server/routes/startup_data.py:68]: I am unsure if we let Notification stand on its own or merge it into User data

Comment (al-yanna) [reply launchpad_backend/launchpad_server/routes/startup_data.py:70]: I think that the current schema is fine as is! There's not much of a difference between keeping them separate and merging them together + User data is less polluted if we keep Notifications as its own collection.

File: launchpad_backend/launchpad_server/routes/startup_data.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    "index": "userId", 
+    "records": [{
+        "userId": 1,
+        "email": "michael.cera@yahoo.com", 
+        "password": "superbad", 
+        "firstName": "Mikey", 
+        "lastName": "C", 
+        "year": "1st/2nd/3rd or Graduated", 
+        "program": "Computer Science",
+        "address": "address",  # If address is not specified for a record, do not include this key-value pair in the dictionary
```
Comment (al-yanna) [inline launchpad_backend/launchpad_server/routes/startup_data.py:28]: Would this be cleaner?: 
```json
"address":  {
   "street": "123 example st",
   "postalCode": "000 000",
   "province": "ON",
}
```

File: launchpad_backend/launchpad_server/routes/index.py [hunk 3]
```diff
@@ -13,11 +17,38 @@
...
+            insertList.append(record)
+
+        collection.insert_many(insertList)
+        collection.create_index([(data["index"], pymongo.ASCENDING)], unique=True) # Enforces primary key like restrcitions
+
+setup_db() # Resets db to startup_data everytime backend is saved or ran
+
+
+@app.route("/applications")
+def check_db():
```
Comment (al-yanna) [inline launchpad_backend/launchpad_server/routes/index.py:36]: `check_db()` is already defined under the `/check_db` route so change this function name to something else.

When I try to run this I get the following error:
`AssertionError: View function mapping is overwriting an existing endpoint function: check_db`

**Review by al-yanna — CHANGES_REQUESTED** [review]

**Review by al-yanna — APPROVED** [review]

**Pull request merged**